README.md

```diff
@@ -273,6 +273,97 @@ manually).  One way to do this, on Linux, is to ensure that the
 
 (or whatever directory `libmlpack.so` is installed in.)
 
+#### Building with Visual Studio 2019 
+
+This project can be build with the latest version of MS Visual Studio, given you have CMake intergration via the [C++ CMake toools for Windows](https://docs.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=msvc-160). To open the CMake project with Visual Studio, select "File->Open->CMake" in the top menu, followed by selecting the root `CMakeLists.txt` located in mlpack's root directory.
+
+In order to allow Visual Studio to configure the CMake project, the CMake configuration json will have to be editted to provide the [relevant options](#4-building-mlpack-from-source) needed to find all the dependencies. The options that you must provide to Visual Studio's CMake are:
+
+* `ARMADILLO_INCLUDE_DIR`
+* `ARMADILLO_LIBRARY`
+* `BOOST_ROOT`
+* `CEREAL_INCLUDE_DIR`
+* `BLAS_LIBRARIES`
+* `LAPACK_LIBRARIES`
+
+The CMake configuration json can be editted in Visual Studio by right clicking the root `CMakeLists.txt` in the project view, selecting **CMake settings for mlpack** and finally clicking on **edit JSON**. Adding new CMake option can be done by adding object fields with the following format to the `variables` array in the `CMakeSettings.json`:
+
+```json
+{
+    "name": "options_name_string",
+    "value": "options_value_string",
+    "type" : "{BOOL|FILEPATH|PATH|STRING}"
+}
+```
+Here is an example of a complete `CMakeSettings.json` needed to build mlpack on Visual Studio:
+
+<details>
+<summary>Click to see the example CMakeSettings.json</summary>
+
+```json
+{
+  "configurations": [
+    {
+      "name": "x64-Debug (default)",
+      "generator": "Ninja",
```

Comment:
Is Ninja generally available in MSVC 2019, or should we use the MSVC generator or something here by default?

Reply:
Ninja is the default generator for CMake projects in VS2019 I believe. The only thing I changed in that json file was the `variables` array.

Reply:
Ahh, perfect, I didn't know that. :+1:  Thanks for the clarification!